bigtop-packages/src/common/hive/patch7-log4j2-2.16.0.diff

```diff
@@ -0,0 +1,167 @@
+     <maven.surefire.plugin.version>2.18.1</maven.surefire.plugin.version>
+     <maven.war.plugin.version>2.4</maven.war.plugin.version>
+     <maven.dependency.plugin.version>2.8</maven.dependency.plugin.version>
+@@ -143,7 +143,7 @@
+     <hamcrest.version>1.1</hamcrest.version>
+     <hbase.version>1.1.1</hbase.version>
+     <!-- required for logging test to avoid including hbase which pulls disruptor transitively -->
+-    <disruptor.version>3.3.0</disruptor.version>
++    <disruptor.version>3.3.7</disruptor.version>
+     <hikaricp.version>2.5.1</hikaricp.version>
+     <!-- httpcomponents are not always in version sync -->
+     <httpcomponents.client.version>4.4</httpcomponents.client.version>
+@@ -173,7 +173,7 @@
+     <kryo.version>3.0.3</kryo.version>
+     <libfb303.version>0.9.3</libfb303.version>
+     <libthrift.version>0.9.3</libthrift.version>
+-    <log4j2.version>2.6.2</log4j2.version>
++    <log4j2.version>2.16.0</log4j2.version>
+     <opencsv.version>2.3</opencsv.version>
+     <orc.version>1.3.4</orc.version>
+     <mockito-all.version>1.9.5</mockito-all.version>
+diff --git a/ql/pom.xml b/ql/pom.xml
+index 3677dc843e..2cdcac43aa 100644
+--- a/ql/pom.xml
++++ b/ql/pom.xml
+@@ -154,6 +154,11 @@
+       <artifactId>log4j-1.2-api</artifactId>
+       <version>${log4j2.version}</version>
+     </dependency>
++    <dependency>
++      <groupId>org.apache.logging.log4j</groupId>
++      <artifactId>log4j-core</artifactId>
++      <version>${log4j2.version}</version>
++    </dependency>
+     <dependency>
+       <groupId>org.apache.logging.log4j</groupId>
+       <artifactId>log4j-slf4j-impl</artifactId>
+diff --git a/ql/src/java/org/apache/hadoop/hive/ql/log/NoDeleteRollingFileAppender.java b/ql/src/java/org/apache/hadoop/hive/ql/log/NoDeleteRollingFileAppender.java
+index be32f06e43..5d10233054 100644
+--- a/ql/src/java/org/apache/hadoop/hive/ql/log/NoDeleteRollingFileAppender.java
++++ b/ql/src/java/org/apache/hadoop/hive/ql/log/NoDeleteRollingFileAppender.java
+@@ -27,7 +27,7 @@
+ import org.apache.log4j.Layout;
+ import org.apache.log4j.helpers.CountingQuietWriter;
+ import org.apache.log4j.helpers.LogLog;
+-import org.apache.log4j.helpers.OptionConverter;
++import org.apache.logging.log4j.core.util.OptionConverter;
+ import org.apache.log4j.spi.LoggingEvent;
+ 
+ public class NoDeleteRollingFileAppender extends FileAppender {
+diff --git a/service/pom.xml b/service/pom.xml
+index aef22881fd..c247d38717 100644
+--- a/service/pom.xml
++++ b/service/pom.xml
+@@ -81,7 +81,12 @@
+           </exclusion>
+         </exclusions>
+    </dependency>
+-    <!-- used by thrift generated code -->
++    <dependency>
++      <groupId>com.lmax</groupId>
++      <artifactId>disruptor</artifactId>
++      <version>${disruptor.version}</version>
++    </dependency>
++   <!-- used by thrift generated code -->
+     <dependency>
+       <groupId>commons-lang</groupId>
+       <artifactId>commons-lang</artifactId>
```

Comment:
same as jdbc/pom.xml, hive-exec does not directly depend on disruptor. It is via hive-llap-server. I think we can omit the change of service/pom.xml.

```
[INFO] org.apache.hive:hive-service:jar:2.3.6
...
[INFO] +- org.apache.hive:hive-llap-server:jar:2.3.6:compile
...
[INFO] |  +- org.apache.hbase:hbase-server:jar:1.5.0:compile
...
[INFO] |  |  \- com.lmax:disruptor:jar:3.3.0:compile
```